tooling/minitest/src/tests/trait_object.rs

```diff
@@ -402,6 +386,37 @@ fn ill_lookup_wrong_ty() {
     );
 }
 
+/// A VTableLookup only works on trait methods that are defined by the particular vtable.
+#[test]
+fn ill_lookup_wrong_method() {
+    let mut p = ProgramBuilder::new();
+
+    let t_builder = p.declare_trait();
+    let trait_name = p.finish_trait(t_builder);
+    let v_builder = p.declare_vtable(trait_name, Size::ZERO, Align::ONE);
+    let vtable_name = p.finish_vtable(v_builder);
+    let mut t_builder = p.declare_trait();
+    let meth2 = t_builder.declare_method();
+    let _trait2_name = p.finish_trait(t_builder);
+
+    let f = {
+        let mut f = p.declare_function();
+        let y = f.declare_local_with_ty(Type::Ptr(PtrType::FnPtr));
+        f.storage_live(y);
+        // Ill formed since the vtable doesn't declare the meth2.
+        // (Actually it might still work, since both miniutil and minimize use low entropy method indices).
```

Comment:
What do you mean by this?

Reply:
It is only checked that some method with the same name exists. However, there are a bunch of homonyms of the method because miniutil uses monotonically increasing numbers and minimize indices.

Reply:
But we control the entire testcase and it's not very big so I fail to see how that's relevant...? Also the test does report "illformed" so how does it make sense to say "it might still work" while at the same time asserting that it does not work?